src/main/java/com/softserve/academy/spaced/repetition/controller/CourseCommentController.java

```diff
@@ -34,70 +35,60 @@
                                                          @PathVariable Long courseId) throws NotAuthorisedUserException {
         LOGGER.debug("Added comment to course with id: {}", courseId);
         CourseComment courseComment = courseCommentService
                 .addCommentForCourse(courseId, replyToCommentDTO.getCommentText(), replyToCommentDTO.getParentCommentId());
         Link selfLink = linkTo(methodOn(CourseCommentController.class)
-                .getCommentByCourse(categoryId, courseId, courseComment.getId())).withSelfRel();
+                .getCommentByCourse(courseId, courseComment.getId())).withSelfRel();
         CommentDTO courseCommentDTO = DTOBuilder.buildDtoForEntity(courseComment, CommentDTO.class, selfLink);
         return new ResponseEntity<>(courseCommentDTO, HttpStatus.CREATED);
     }
 
     @Auditable(action = AuditingAction.EDIT_COMMENT_FOR_COURSE)
-    @PutMapping(value = "/api/category/{categoryId}/courses/{courseId}/comments/{courseCommentId}")
-    @PreAuthorize(value = "@accessToUrlService.hasAccessToUpdateCommentForCourse(#courseCommentId)")
+    @PutMapping(value = "/{courseCommentId}")
     public ResponseEntity<CommentDTO> updateComment(@Validated(Request.class) @RequestBody String courseContentComment,
-                                                    @PathVariable Long categoryId,
                                                     @PathVariable Long courseId,
                                                     @PathVariable Long courseCommentId) {
         LOGGER.debug("Updated courseComment with id: {}", courseCommentId);
         CourseComment courseComment = courseCommentService.updateCommentById(courseCommentId, courseContentComment);
         Link selfLink = linkTo(methodOn(CourseCommentController.class)
-                .getCommentByCourse(categoryId, courseId, courseCommentId)).withSelfRel();
+                .getCommentByCourse(courseId, courseCommentId)).withSelfRel();
         CommentDTO courseCommentDTO = DTOBuilder.buildDtoForEntity(courseComment, CommentDTO.class, selfLink);
         return new ResponseEntity<>(courseCommentDTO, HttpStatus.OK);
     }
 
     @Auditable(action = AuditingAction.DELETE_COMMENT_FOR_COURSE)
-    @DeleteMapping(value = "/api/category/{categoryId}/courses/{courseId}/comments/{courseCommentId}")
-    @PreAuthorize(value = "@accessToUrlService.hasAccessToDeleteCommentForCourse(#courseCommentId)")
-    public ResponseEntity deleteComment(@PathVariable Long courseCommentId,
-                                        @PathVariable Long courseId,
-                                        @PathVariable Long categoryId) {
-        LOGGER.debug("Deleted comment with id:{} for course with id: {}", courseCommentId, courseId);
+    @DeleteMapping(value = "/{courseCommentId}")
+    public ResponseEntity deleteComment(@PathVariable Long courseCommentId) {
```

Comment:
I don't know exactly, but i think that this method should take `courseId` and `commentId`.
Inside of service method, we can get some `Course` by id. After it, we can get all comments of that course and delete.
I'm not sure, that this approach is better than current ( we just delete comment by Id ), but think it's more right way.